models/bridge/zksync/bridge_zksync_native_flows.sql

```diff
@@ -0,0 +1,174 @@
+{{ config(
+    schema = 'bridge_zksync_native',
+    alias = 'flows',
+    partition_by = ['block_month'],
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['block_date', 'blockchain', 'tx_hash', 'evt_index'],
+    post_hook='{{ expose_spells(\'["zksync"]\',
+                                "sector",
+                                "bridge",
+                                \'["lgingerich"]\') }}'
+    )
+}}
+
+WITH bridge_events as (
+    SELECT
+         block_time
+        ,CAST(DATE_TRUNC('day', block_time) as date) as block_date
+        ,CAST(DATE_TRUNC('month', block_time) as date) as block_month
+        ,block_number
+        ,tx_hash
+        ,evt_index
+        ,sender_address
+        ,receiver_address
+        ,bridged_token_address
+        ,bridged_token_amount_raw
+        ,source_chain_id
+        ,destination_chain_id
+        ,source_chain_name
+        ,destination_chain_name
+        ,tx_from
+        ,tx_to
+    FROM (
+        -- Deposit ETH from Ethereum to zkSync Era
+        SELECT
+             npr.evt_block_time as block_time
+            ,npr.evt_block_number as block_number
+            ,npr.evt_tx_hash as tx_hash
+            ,npr.evt_index
+            ,et."from" as sender_address
+            ,COALESCE(d.to, zt.to) as receiver_address -- d.to is null if there is no matching ERC20 deposit tx hash. This is used to handle the logic of an 'if statement'.
+            ,0x0000000000000000000000000000000000000000 as bridged_token_address
+            ,CAST(JSON_EXTRACT_SCALAR(npr.transaction, '$.reserved[0]') as UINT256) as bridged_token_amount_raw
+            ,UINT256 '1' as source_chain_id
+            ,UINT256 '324' as destination_chain_id
+            ,'ethereum' as source_chain_name
+            ,'zksync' as destination_chain_name
+            ,et."from" as tx_from
+            ,et.to as tx_to
+        FROM {{ source('ethereum', 'transactions') }} et
+        INNER JOIN {{ source('zksync_v2_ethereum', 'DiamondProxy_evt_NewPriorityRequest') }} npr ON et.hash = npr.evt_tx_hash
```

Comment:
would an incremental filter on this join make sense too? only if the decoded table is event/call driven so it has lots of rows. if it's a smaller table, no worries

Reply:
I'm not great with incremental logic in dbt yet, so not exactly sure what you're suggesting, but this is a decoded table with lots of rows so it may make sense to add.

If you don't mind, I'll let you make a suggestion in the code

Reply:
sure, will do